si-ap-idff/si_ap_idff/si_ap_idff.py

```diff
@@ -0,0 +1,111 @@
+"""SI-AP-IDFF Soft IOC."""
+
+import os as _os
+import sys as _sys
+import signal as _signal
+import pcaspy as _pcaspy
+import pcaspy.tools as _pcaspy_tools
+
+from siriuspy import util as _util
+from siriuspy.envars import VACA_PREFIX as _vaca_prefix
+
+from siriuspy.idff.main import App as _App
+
+
+INTERVAL = 0.1
+STOP_EVENT = False
+
+
+def _stop_now(signum, frame):
+    _ = frame
+    print(_signal.Signals(signum).name+' received at '+_util.get_timestamp())
+    _sys.stdout.flush()
+    _sys.stderr.flush()
+    global STOP_EVENT
+    STOP_EVENT = True
+
+
+def _attribute_access_security_group(server, dbase):
+    for k, val in dbase.items():
+        if k.endswith(('-RB', '-Sts', '-Cte', '-Mon')):
+            val.update({'asg': 'rbpv'})
+    path_ = _os.path.abspath(_os.path.dirname(__file__))
+    server.initAccessSecurityFile(path_ + '/access_rules.as')
+
+
+class _PCASDriver(_pcaspy.Driver):
+
+    def __init__(self, app):
+        """Initialize driver."""
+        super().__init__()
+        self.app = app
+        self.app.add_callback(self.update_pv)
+
+    def read(self, reason):
+        value = self.app.read(reason)
+        if value is None:
+            return super().read(reason)
+        return value
+
+    def write(self, reason, value):
+        """Write IOC pvs acording to main application."""
+        if self.app.write(reason, value):
+            return super().write(reason, value)
+        return False
+
+    def update_pv(self, pvname, value, **kwargs):
+        """Update PV."""
+        self.setParam(pvname, value)
+        self.updatePV(pvname)
+
+
+def run(idname):
+    """Run main module function."""
+    # define abort function
+    _signal.signal(_signal.SIGINT, _stop_now)
+    _signal.signal(_signal.SIGTERM, _stop_now)
+
+    # configure log file
+    _util.configure_log_file()
+
+    # define IOC, init pvs database and create app object
+    _version = '1.0.0'
```

Comment:
is this how we have been setting the IOC version? are we not taking it from git commit hash?

Reply:
the commit hash only works for local installation. This is a known bug which will take me a while to work on... All of our Version-Cte PVs are now unused. I've set the version to 1.0.0 here just to get around this, but the idea is to fix the commit hash and use it in all IOCs.